Prep for 1.0 release

SUPERCILEX opened the pull request:
fixes #154

SUPERCILEX commented:
I really hate tests sometimes 😆

@samtstern 
Anyway, this PR is now a little bigger than I originally intended:
- Removes all deprecated APIs (which is just the `AppSettingsDialog` constructors and negative button listener method)
  - Also refactored how we handle clicks and contexts
- Fixed tests failing because of AAPT2
  - I needed to do this because AAPT1 was failing to compile a string for some reason.

SUPERCILEX commented:
Also:
- Removed `BasicActivity`: it says it's for tests, but I don't see it used in programmatic tests and it appears to be a subset of `MainActivity` anyway
- Use 3.0 configuration

SUPERCILEX commented on easypermissions/build.gradle:
```diff
@@ -20,7 +20,12 @@ android {
-    compile "com.android.support:appcompat-v7:$support_library_version"
+    implementation "com.android.support:appcompat-v7:$support_library_version"
+    api "com.android.support:support-compat:$support_library_version"
+    api "com.android.support:support-fragment:$support_library_version"
```
When I removed `appcompat-v7` from the demo project, I couldn't get it to compile without these api declarations.

SUPERCILEX commented:
@samtstern Ok, this PR has grown too big so I've rebased all my commits such that they can be merged individually (instead of the usual squashing). I could also try to create a PR per commit, what do you think?

Since my last comment, I've fixed a bug where the `AppSettingsDialog` would use system default colors instead of picking the dev's app colors.

samtstern commented on easypermissions/src/main/java/pub/devrel/easypermissions/AppSettingsDialog.java:
```diff
@@ -65,35 +61,37 @@ private AppSettingsDialog(Parcel in) {
 
-    void setNegativeListener(DialogInterface.OnClickListener negativeListener) {
-        mNegativeListener = negativeListener;
+    private static Context parseContext(Object activityOrFragment) {
```
Could we make this method something we call just once (at `build()` time for example?) and then save the `Context` in a variable rather than re-casting `activityOrFragment` every time we want to use it?

samtstern commented on easypermissions/src/main/java/pub/devrel/easypermissions/AppSettingsDialog.java:
```diff
@@ -65,35 +61,37 @@ private AppSettingsDialog(Parcel in) {
-
-    void setContext(Context context) {
-        mContext = context;
+    static AppSettingsDialog parse(Intent intent, Activity activity) {
```
nit: I would call this method either `getInstance()` or `fromIntent()`

samtstern commented:
@SUPERCILEX this PR is big but not problematically so, I think we can try to keep it together.  Thank you for sending it!

SUPERCILEX commented:
@samtstern SGTM! I rebased everything because I thought you could compare two commits coming from the same HEAD, but apparently not so the diff is broken, sorry! 😊 I added back the `mContext` fields in the new 6aa514e and renamed that method to `fromIntent`.

While working on `AppSettingsDialog`, I noticed that we had a bunch of unnecessary `RequiresApi` annotations back from when our min sdk was 9 so I fixed that in e1c9301. And while working on _that_ 😄, I noticed that we were using the framework `Fragment` even if the calling activity was a descendant of `AppCompatActivity` so I fixed that in  a497251. Whew! I think I'm done now. 😆

samtstern commented:
@SUPERCILEX thanks for all the hard work here!  This LGTM, I am going to merge it.
